ERCS/erc-7715.md

```diff
@@ -0,0 +1,480 @@
+```
+
+- `grantedPermissions` is a list of permissions that were granted. The wallet MUST grant `required` permissions if the request was approved.
+- `expiry` is the expiry time that was granted. The wallet MAY grant a different expiry time than the one that the DApp requested.
+- `factory` and `factoryData` are as defined in [ERC-4337](./eip-4337.md). They are either both specified, or none. If the account has not yet been deployed, the wallet MUST return `factory` and `factoryData`, and the DApp MUST deploy the account by calling the `factory` contract with `factoryData` as the calldata.
+- `permissionsContext` is an opaque string that identifies the session. It MAY be the `context` as defined in [ERC-7679](./eip-7679.md) and [ERC-7710](./eip-7710.md). See “Rationale” for details.
+
+If the request is malformed or the wallet is unable/unwilling to grant permissions, wallet MUST return an error with a code as defined in [ERC-1193](./eip-1193.md).
+
+`wallet_grantPermissions` response example:
+
+```tsx
+{
+  grantedPermissions: [
+    {
+      type: 'erc20-spending-limit',
+      data: {
+        erc20Address: '0x...',
+        limit: '0x...',
+      },
+    },
+    {
+      type: 'gas-limit',
+      data: {
+        limit: '0x...',
+      },
+    }
+  ],
+
+  expiry: 1577840461,
+
+  permissionsContext: '0x...',
+}
+```
+
+### Signer & Permission Types
+
+In this ERC, we specify a list of signers and permissions that we expect to be commonly used.
+
+This ERC does not specify an exhaustive list of signer or permission types, since we expect more signer/permission types to be developed as wallets get more advanced. A signer/permission type is valid as long as both the DApp and the wallet are willing to support it.
+
+However, if two signers or two permissions share the same type name, a DApp could request with one type of signer/permission while the wallet grants another. Therefore, it’s important that no two signers or two permissions share the same type. Therefore, new signer or permission types should be specified in ERCs, either in this ERC as an amendment or in another ERC.
+
+#### Signers
+
+```tsx
+// A wallet is the signer for these permissions
+// `data` is not necessary for this signer type as the wallet is both the signer and grantor of these permissions
+type WalletSigner = {
+  type: "wallet";
+  data: {};
+};
+
+// A signer representing a single key.
+// `id` is a did:key identifier and can therefore represent both Secp256k1 or Secp256r1 keys, among other key types.
+type KeySigner = {
+  type: "key";
+  data: {
+    id: string;
+  };
+};
+
+// A signer representing a multisig signer.
+// Each element of `ids` is a did:key identifier just like the `key` signer.
+type MultiKeySigner = {
+  type: "keys";
+  data: {
+    ids: string[];
+  };
+};
+
+// An account that can be granted with permissions as in ERC-7710.
+type AccountSigner = {
+  type: "account";
+  data: {
+    id: `0x${string}`;
+  };
+};
+```
+
+### Permissions Types
```

Comment:
I do not think we should have **any** canonical permission types in this EIP to keep this proposal tight. That said, I have dropped requested changes to improve the documentation of these examples.

Two general comments:
1. I think we should prefer `value` over `amount` to be more obviously connected to `msg.value` and ERC20 spec.
2. I think all of these a CAIP-2 `chainId` to be more specific about where these permissions can apply? I do not see a notion of chain-specificity in the top-level session definition, but I think it's better to apply it at the permissions layer anyways for modularity of request permissions across multiple networks simultaneously. Are we comfortable scaling multi-chain permissions by just requesting the same object but with a different `chainId` parameter?